Src/ILGPU/IR/Construction/Structures.cs

```diff
@@ -80,6 +84,34 @@ partial class IRBuilder
                 else
                 {
                     instanceBuilder.Add(fieldValue);
+
+                    // If the field value we just added was the start of a fixed buffer,
+                    // copy the remaining elements, byte by byte.
+                    if (field.GetCustomAttribute<FixedBufferAttribute>() != null)
```

Comment:
QUESTION: Is this the best way to detected a fixed buffer?

Reply:
In general, there is no guarantee that all .Net compatible compilers attach this attribute. What we could to is to check whether the native `size` of a type enforces a custom "padding".

Reply:
Changed to compare field size.